src/js/actions/libraries.js

```diff
@@ -335,17 +337,20 @@ define(function (require, exports) {
             return Promise.resolve();
         }
 
-        var docRef = docAdapter.referenceBy.id(currentDocument.id),
-            location = { x: 100, y: 100 },
-            representation = element.getPrimaryRepresentation();
-
-        return Promise.fromNode(function (cb) {
-            representation.getContentPath(cb);
-        }).then(function (path) {
-            var placeObj = libraryAdapter.placeElement(docRef, element, path, location);
-
-            return descriptor.playObject(placeObj);
-        });
+        return Promise
+                .fromNode(function (cb) {
+                    element.getPrimaryRepresentation().getContentPath(cb);
+                })
+                .bind(this)
+                .then(function (path) {
+                    var docRef = docAdapter.referenceBy.id(currentDocument.id),
+                        placeObj = libraryAdapter.placeElement(docRef, element, path, location);
+
+                    return descriptor.playObject(placeObj);
+                })
+                .then(function () {
+                    this.flux.actions.documents.updateDocument();
```

Comment:
Yes, this is very important. `updateDocument` is extremely slow in large documents. If we need Photoshop to emit an event with the layer IDs and positions of the placed document, we should file a high-priority Watson bug for that.